compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1035,6 +1035,15 @@ pub enum Rvalue<'tcx> {
     /// initialized but its content as uninitialized. Like other pointer casts, this in general
     /// affects alias analysis.
     ShallowInitBox(Operand<'tcx>, Ty<'tcx>),
+
+    /// A CopyForDeref is equivalent to a read from a place at the
+    /// codegen level, but is treated specially by drop elaboration. When such a read happens, it
+    /// is guaranteed (via nature of deref_seperator) that the only use of the returned value is a deref operation, immediately
```

Comment:
What is `deref_seperator`? Please mention the file/module where it is defined.

Reply:
It's a mir-opt I added the path to it in the comment